File: fix_gallery.py
import re

path = r'C:\Users\A-MATRIX\Downloads\fario-ecommerce-app-main\fario-ecommerce-app-main\components\home\HomeGallery.tsx'

with open(path, 'rb') as f:
    content = f.read()

# The BEL character \x07 replaced the backtick+a from `absolute
# The corrupted pattern is: cls={\x07bsolute inset-0 w-full h-full object-cover transition-transform duration-700 ease-out }
old = b'cls={\x07bsolute inset-0 w-full h-full object-cover transition-transform duration-700 ease-out }\n            px={0}\n            />'
new = b"cls={`absolute inset-0 w-full h-full object-cover transition-transform duration-700 ease-out ${isHovered ? 'scale-110' : 'scale-150 grayscale-[0.3]'}`} \n                px={0}\n            />"

if old in content:
    content = content.replace(old, new)
    print('Fix applied successfully!')
else:
    print('Pattern not found, trying alternative...')
    # Try without the surrounding context
    old2 = b'cls={\x07bsolute'
    new2 = b'cls={`absolute inset-0 w-full h-full object-cover transition-transform duration-700 ease-out ${isHovered ? \'scale-110\' : \'scale-150 grayscale-[0.3]\'}`}'
    idx = content.find(b'cls={\x07')
    if idx != -1:
        # Find the end of this prop (next newline after the })
        end = content.find(b'px={0}', idx)
        replacement = b"cls={`absolute inset-0 w-full h-full object-cover transition-transform duration-700 ease-out ${isHovered ? 'scale-110' : 'scale-150 grayscale-[0.3]'}`} \n                px={0}"
        content = content[:idx] + replacement + content[end + len(b'px={0}'):]
        print('Alternative fix applied!')
    else:
        print('ERROR: Could not find pattern to fix!')

with open(path, 'wb') as f:
    f.write(content)

print('File saved.')

# Verify
with open(path, 'r', encoding='utf-8') as f:
    lines = f.readlines()
print('Line 74:', repr(lines[73]))

File: fix_gallery.js
const fs = require('fs');
const path = 'C:/Users/A-MATRIX/Downloads/fario-ecommerce-app-main/fario-ecommerce-app-main/components/home/HomeGallery.tsx';

let content = fs.readFileSync(path, 'utf8');

// Fix 1: The corrupted cls prop on line 74
// The \u0007 (BEL char) replaced the backtick+a of `absolute
content = content.replace(
    /cls=\{[\x07\u0007]bsolute inset-0 w-full h-full object-cover transition-transform duration-700 ease-out \}/,
    "cls={`absolute inset-0 w-full h-full object-cover transition-transform duration-700 ease-out ${isHovered ? 'scale-110' : 'scale-150 grayscale-[0.3]'}`}"
);

// Fix 2: The corrupted animate background on lines 122-123
// The backslash-newline form corrupted the template literal
content = content.replace(
    /animate=\{\{ background: \\\nadial-gradient\(ellipse at bottom right, \\ 0%, transparent 70%\)\\ \}\}/,
    'animate={{ background: `radial-gradient(ellipse at bottom right, ${activeCat.accent} 0%, transparent 70%)` }}'
);

fs.writeFileSync(path, content, 'utf8');
console.log('Fixed! Lines changed:');

// Verify fix
const lines = content.split('\n');
console.log('Line 74:', lines[73]);
console.log('Line 122:', lines[121]);
